scripts/nix-profile-daemon.fish.in

```diff
@@ -0,0 +1,35 @@
+# Only execute this file once per shell.
```

Comment:
I'm not very happy that we now have 4 nearly identical files: nix-profile-daemon.sh.in, nix-profile.sh.in, nix-profile-daemon.fish.in and nix-profile.fish.in. We should at least try to consolidate the daemon/non-daemon variants into one.

Reply:
I could've sworn I remembered seeing an old PR to this effect at some point, but I didn't have any luck searching for it just now.

It bugs me, too (especially since there are inevitably other shells that would like the same treatment, as in https://github.com/NixOS/nix/issues/6239). There's some old discussion of a few approaches to generating the fish from shell in https://github.com/NixOS/nix/issues/440. 

I can imagine writing a shell script with per-format output functions to generate N formats to handle it on that dimension, though it'd probably be better for that to happen after merging nix-profile and nix-profile-daemon if that's already desirable for its own reasons. Both feel like they'd be a big increase in scope for this PR, though.

Reply:
Perhaps https://github.com/NixOS/nix/issues/7027 would sufficiently address it?

Reply:
We decided to punt this issue for another PR.